ownership-chain/runtime/laos/src/configs/block_rewards_source.rs

```diff
@@ -1,5 +1,106 @@
-use crate::{Runtime, RuntimeEvent};
+use crate::{Balances, Runtime};
+use frame_support::weights::Weight;
+use pallet_block_rewards_source::{BalanceOf, WeightInfo};
+use pallet_parachain_staking::PayoutCollatorReward;
+use sp_runtime::{traits::Zero, DispatchError};
+use sp_std::marker::PhantomData;
 
 impl pallet_block_rewards_source::Config for Runtime {
-	type RuntimeEvent = RuntimeEvent;
+	type WeightInfo = pallet_block_rewards_source::weights::SubstrateWeight<Runtime>;
+	type Currency = Balances;
+}
+pub struct BlockRewardsSourceAdapter<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime: pallet_parachain_staking::Config + pallet_block_rewards_source::Config>
+	PayoutCollatorReward<Runtime, BalanceOf<Runtime>> for BlockRewardsSourceAdapter<Runtime>
+{
+	fn payout_collator_reward(
+		_round_index: pallet_parachain_staking::RoundIndex,
+		collator_id: Runtime::AccountId,
+		amount: pallet_block_rewards_source::BalanceOf<Runtime>,
+	) -> Weight {
+		match pallet_block_rewards_source::Pallet::<Runtime>::send_rewards(
+			collator_id,
+			amount.into(),
+		) {
+			Ok(amount) if amount.is_zero() => Weight::zero(),
```

Comment:
I don't really have a strong opinion here, just wondering if there is more efficient way of doing it. as I said, it might be ok to not care about one read operation, but if we can fix that edge case, would be nice.

Maybe returning weight from the `send_rewards` could be a solution

Reply:
it seems reasonable to me return `send_rewards`, left a comment to see if it can be improved in the future